neqo-client/src/main.rs

```diff
@@ -528,29 +631,46 @@ impl<'a> Handler<'a> {
         }
         true
     }
+}
+
+fn to_headers(values: &[impl AsRef<str>]) -> Vec<Header> {
+    values
+        .iter()
+        .scan(None, |state, value| {
+            if let Some(name) = state.take() {
+                *state = None;
```

Comment:
Do you need to assign after having called `.take()`?

Reply:
Pleas see [here](https://github.com/mozilla/neqo/blob/main/neqo-client/src/main.rs#L621).
This is actually the original code, but I'll think about this comment.